### PR TITLE
[chore] fix exhaustive lint statsd receiver

### PR DESCRIPTION
**Pull request by jaydeluca**
**Description:** <Describe what has changed.>
Related #23266

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: receiver/statsdreceiver/config.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (c *Config) Validate() error {
 		}
 
 		switch eachMap.StatsdType {
-		case protocol.TimingTypeName, protocol.TimingAltTypeName, protocol.HistogramTypeName:
+		case protocol.TimingTypeName, protocol.TimingAltTypeName, protocol.HistogramTypeName, protocol.CounterTypeName, protocol.GaugeTypeName:
```
Comment (codeboten) [inline receiver/statsdreceiver/config.go:42]: Is this an expected change? before this, `protocol.CounterTypeName` & `protocol.GaugeTypeName` would produce an error. After this change they will no longer do so.

Pinging code owners @jmacd & @dmitryax for confirmation here

File: receiver/statsdreceiver/config.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func (c *Config) Validate() error {
 		}
 
 		switch eachMap.ObserverType {
-		case protocol.GaugeObserver, protocol.SummaryObserver, protocol.HistogramObserver:
+		case protocol.GaugeObserver, protocol.SummaryObserver, protocol.HistogramObserver, protocol.DefaultObserverType:
```
Comment (codeboten) [inline receiver/statsdreceiver/config.go:53]: Same comment as above here

**Review by codeboten — CHANGES_REQUESTED** [review]
Marking this as "request changes" to prevent accidental merge as I'm waiting for confirmation from code owners before moving the PR forward.

File: receiver/statsdreceiver/config.go (hunk 1 above)
Comment (jmacd) [reply receiver/statsdreceiver/config.go:42]: This doesn't look right. Why would a linter complain when there's a default branch? 
The default branch is correct but the error message is misleading. "statsd_type is not a supported mapping" could be replaced with "statsd_type is not a supported mapping for histogram and timing metrics".

I see this change was applied in another PR, so it's pointless to review here? https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24046

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 3]
```diff
@@ -168,6 +168,8 @@ func (p *StatsDParser) Initialize(enableMetricType bool, isMonotonicCounter bool
 		case TimingTypeName, TimingAltTypeName:
 			p.timerEvents.method = eachMap.ObserverType
 			p.timerEvents.histogramConfig = expoHistogramConfig(eachMap.Histogram)
+		case CounterTypeName:
+		case GaugeTypeName:
```
Comment (jmacd) [inline receiver/statsdreceiver/protocol/statsd_parser.go:172]: I personally object to this level of "linting". Can these two branches be a single `default` branch instead? (What is actually improving here?)

File: receiver/statsdreceiver/protocol/statsd_parser.go [hunk 4]
```diff
@@ -258,6 +260,8 @@ func (p *StatsDParser) observerCategoryFor(t MetricType) ObserverCategory {
 		return p.histogramEvents
 	case TimingType:
 		return p.timerEvents
+	case CounterType:
+	case GaugeType:
```
Comment (jmacd) [inline receiver/statsdreceiver/protocol/statsd_parser.go:264]: Same statement: this was intentional and I'm not sure what the linter is adding.

**Review by jmacd — CHANGES_REQUESTED** [review]

File: receiver/statsdreceiver/config.go (hunk 1 above)
Comment (codeboten) [reply receiver/statsdreceiver/config.go:42]: > This doesn't look right. Why would a linter complain when there's a default branch?

This is a feature of the exhaustive linter: `ExplicitExhaustiveSwitchFlag` https://pkg.go.dev/github.com/nishanths/exhaustive#pkg-constants

Comment (jaydeluca) [reply receiver/statsdreceiver/config.go:42]: > I see this change was applied in another PR, so it's pointless to review here? https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24046

Yea sorry about that, I didn't see that other PR

**Pull request closed without merging**